bucket/simple-dnscrypt.json

```
@@ -0,0 +1,68 @@
+{
+    "version": "0.7.1",
+    "description": "An easy to use GUI for dnscrypt-proxy",
+    "homepage": "https://simplednscrypt.org/",
+    "license": "MIT",
+    "suggest": {
+        "vcredist2017": "extras/vcredist2017"
+    },
+    "architecture": {
+        "64bit": {
+            "url": "https://github.com/bitbeans/SimpleDnsCrypt/releases/download/0.7.1/SimpleDNSCrypt64.msi",
+            "hash": "5BA0C94774ACD7ADE18446B4351E7D28A948C490AEFC50DC6C08C13551F19F7F",
+            "post_install": [
+                "if (Test-Path \"$dir\\SimpleDNSCrypt64.msi\") {Remove-Item \"$dir\\SimpleDNSCrypt64.msi\" | Out-Null}"
+            ]
+        },
+        "32bit": {
+            "url": "https://github.com/bitbeans/SimpleDnsCrypt/releases/download/0.7.1/SimpleDNSCrypt.msi",
+            "hash": "CAC6B8119AF3DA70A5977A2E73EE5214303990FEDD7ABBC5F247A1E2CF19A663",
+            "post_install": [
+                "if (Test-Path \"$dir\\SimpleDNSCrypt.msi\") {Remove-Item \"$dir\\SimpleDNSCrypt.msi\" | Out-Null}"
+            ]
+        }
+    },
+    "pre_install": [
+        "if (!(is_admin)) { error \"$app requires admin rights to $cmd\"; break }",
+        "",
+        "if (!(Test-Path \"$env:LOCALAPPDATA\\SimpleDnsCrypt\")) {New-Item -Path \"$env:LOCALAPPDATA\" -Name \"SimpleDnsCrypt\" -ItemType \"directory\" -Force | Out-Null}"
+    ],
+    "shortcuts": [
+        [
+            "SimpleDnsCrypt.exe",
+            "Simple DnsCrypt"
+        ]
+    ],
+    "bin": "SimpleDnsCrypt.exe",
+    "persist": [
+        "dnscrypt-proxy\\"
+    ],
+    "pre_uninstall": [
+        "if (!(is_admin)) { error \"$app requires admin rights to $cmd\"; break }",
+        "",
+        "if (Get-Process -Name \"SimpleDnsCrypt\") {Stop-Process -Name \"SimpleDnsCrypt\" | Out-Null}",
+        "if (Get-Service -Name \"dnscrypt-proxy\") {Stop-Service -Name \"dnscrypt-proxy\"| Out-Null; Remove-Service -Name \"dnscrypt-proxy\" | Out-Null}",
+        "if (Test-Path \"$dir\") {Remove-Item \"$dir\" -Force -Recurse | Out-Null}",
+        "if (Test-Path \"$persist_dir\") {Remove-Item \"$persist_dir\" -Force -Recurse | Out-Null}"
+    ],
+    "checkver": {
+        "github": "https://github.com/bitbeans/SimpleDnsCrypt"
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://github.com/bitbeans/SimpleDnsCrypt/releases/download/$version/SimpleDNSCrypt64.msi",
+                "post_install": [
+                    "if (Test-Path \"$dir\\SimpleDNSCrypt64.msi\") {Remove-Item \"$dir\\SimpleDNSCrypt64.msi\" | Out-Null}"
+                ]
+            },
+            "32bit": {
+                "url": "https://github.com/bitbeans/SimpleDnsCrypt/releases/download/$version/SimpleDNSCrypt.msi",
+                "post_install": [
+                    "if (Test-Path \"$dir\\SimpleDNSCrypt.msi\") {Remove-Item \"$dir\\SimpleDNSCrypt.msi\" | Out-Null}"
+                ]
+            }
+        }
+    }
```

```suggestion
    "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://github.com/bitbeans/SimpleDnsCrypt/releases/download/$version/SimpleDNSCrypt64.msi"
            },
            "32bit": {
                "url": "https://github.com/bitbeans/SimpleDnsCrypt/releases/download/$version/SimpleDNSCrypt.msi"
            }
        }
    }
```

`autoupdate` -> `post_install` is only needed if there is something in *post_install* that has to be modified when version changes. (For example, if the program is named something like `someapp-2.1.3.exe`)

If the script has `64bit` -> `post_install`, but has no `autoupdate` -> `64bit` -> `post_install`, then, the existing `64bit` -> `post_install` will **remain unchanged between updates**.

---

Do you mean it is unnecessary to have that same 2 scripts twice? Does that mean I don't need the `post_install` scripts in the `auto_update` section?

---

> Does that mean I don't need the `post_install` scripts in the `autoupdate` section?

Yes. There is no need to have `post_install` scripts in the `autoupdate` section, unless `post_install` contains something has to be changed when "version" changes.